### ISSUE TITLE
Watcher should be Disposable

### ISSUE THREAD

**Issue by DavidANeil** [proposal-signals/proposal-signals#117]
A watcher should be easily disposable, without needing to track which signals it was previous watching.  Using the (currently) stage 3 [Symbol.dispose](https://github.com/tc39/proposal-explicit-resource-management) seems most appropriate.

## Normative Spec Text

Method: `Signal.subtle.Watcher.prototype[Symbol.dispose]()`
1. Invoke `this.unwatch.apply(this, [[signals]])`

**Comment by EisenbergEffect** [comment]
In the meantime, here's how you can do it with the current API:

```js
watcher.unwatch(...Signal.subtle.introspectSources(watcher));
```

**Comment by littledan** [comment]
Right, we could add this as a very small convenience, but it definitely doesn’t require any extra tracking to unwatch all sources, due to introspection. And you can look at the git history to see that we very recently did have such a Symbol.dispose method—I have gone back and forth in my thoughts on whether this is important.
